cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1333,16 +1334,21 @@ private int moveLeadershipInBatch() throws InterruptedException, ExecutionExcept
         if (LOG.isDebugEnabled()) {
           LOG.debug("Tasks in execution: {}", inExecutionTasks());
         }
-        List<ExecutionTask> deadOrAbortingInterBrokerReplicaActions = new ArrayList<>();
+        List<ExecutionTask> deadInterBrokerReplicaTasks = new ArrayList<>();
+        List<ExecutionTask> stoppedInterBrokerReplicaTasks = new ArrayList<>();
         List<ExecutionTask> slowTasksToReport = new ArrayList<>();
         boolean shouldReportSlowTasks = _time.milliseconds() - _lastSlowTaskReportingTimeMs > SLOW_TASK_ALERTING_BACKOFF_TIME_MS;
         for (ExecutionTask task : inExecutionTasks()) {
           TopicPartition tp = task.proposal().topicPartition();
-          if (_stopSignal.get() == FORCE_STOP_EXECUTION) {
-            LOG.debug("Task {} is marked as dead to force execution to stop.", task);
+          if (_stopSignal.get() != NO_STOP_EXECUTION) {
```

Comment:
Prior to this patch, we act on `STOP_EXECUTION`  in other conditions but we handle that in this condition now. Is it intentional?

Reply:
Prior to this PR, only `FORCE_STOP_EXECUTION` (i.e. a force-stop execution request) had a special handling in this loop. Whereas, `STOP_EXECUTION` did not have any handling -- i.e. all tasks in execution were treated like any other in-progress tasks, and were being tracked until the completion (In pre Kafka 2.4.* versions, the role of `STOP_EXECUTION` was just avoid submitting a new batch of execution and wait for the completion of the ongoing batch).

However, the current logic in the initial version of this patch was incomplete. More specifically, we should handle the rollback for gracefully stopped inter-broker replica reassignments. And also we should avoid running a rollback for force-stopped executions -- i.e. they are already handled without rollback.
I updated the handling logic along with comments to explain the reasoning. Thanks for pointing this out!